contracts/cosmwasm-vm/cw-wasm-light-client/src/contract.rs

```diff
@@ -0,0 +1,234 @@
+) -> Result<Response, ContractError> {
+    let data = process_message(deps_mut, env, info, msg)?;
+    let mut response = Response::default();
+    response.data = Some(data);
+    Ok(response)
+}
+
+fn process_message(
+    deps_mut: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Binary, ContractError> {
+    cw_println!(
+        deps_mut.api,
+        "[WasmClient]: Contract Execute Called with {:?}",
+        msg
+    );
+    match msg {
+        ExecuteMsg::VerifyMembership(mut msg) => {
+            cw_println!(deps_mut.api, "[WasmClient]: Verify Membership called");
+            let height = msg.height.revision_height;
+            if msg.proof == "empty".as_bytes() {
+                msg.proof = vec![];
+            }
+            let client_id = CLIENT_ID;
+            let proofs_decoded =
+                MerkleProofs::decode(msg.proof.as_slice()).map_err(ContractError::DecodeError)?;
+            cw_println!(
+                deps_mut.api,
+                "[WasmClient]: Contract Execute Called with Path {:?}",
+                msg.path.key_path
+            );
+            let fullpath = msg.path.key_path[1].clone();
+            cw_println!(deps_mut.api, "[WasmClient]: Full Path is {:?}", fullpath);
+            let path = fullpath.as_bytes().to_vec();
+            let (skip, value) = unwrap_any_type(deps_mut.as_ref(), &msg.value);
+
+            if skip {
+                return to_binary(&ContractResult::success()).map_err(ContractError::Std);
+            }
+
+            let mut res = QueryHandler::verify_membership(
+                deps_mut.as_ref(),
+                client_id,
+                height,
+                msg.delay_time_period,
+                msg.delay_block_period,
+                &proofs_decoded.proofs,
+                &value,
+                &path,
+            );
+            if res.is_err() {
+                res = QueryHandler::verify_membership(
+                    deps_mut.as_ref(),
+                    client_id,
+                    height,
+                    msg.delay_time_period,
+                    msg.delay_block_period,
+                    &proofs_decoded.proofs,
+                    &sha256(value),
+                    &path,
+                );
+            }
+            res.unwrap();
+
+            cw_println!(deps_mut.api, "[WasmClient]: Verify Membership Complete");
+
+            Ok(to_binary(&ContractResult::success()).map_err(ContractError::Std)?)
+        }
+        ExecuteMsg::VerifyNonMembership(msg) => {
+            cw_println!(deps_mut.api, "[WasmClient]: Verify Non Membership Called");
+            let height = msg.height.revision_height;
+            let client_id = CLIENT_ID;
+            let proofs_decoded =
+                MerkleProofs::decode(msg.proof.as_slice()).map_err(ContractError::DecodeError)?;
+            let path = hex::decode(msg.path.key_path.join("")).unwrap();
+
+            QueryHandler::verify_non_membership(
```

Comment:
Why is there no implementation of verification of non membership?

Reply:
non membership verification here is membership verification with blank value